extensions/interactions/NumberWithUnits/directives/oppia-interactive-number-with-units.component.ts

```diff
@@ -13,113 +13,133 @@
+import { NumberWithUnitsObjectFactory } from 'domain/objects/NumberWithUnitsObjectFactory';
+import { NumberWithUnitsRulesService } from './number-with-units-rules.service';
+
+@Component({
+  selector: 'oppia-interactive-number-with-units',
+  templateUrl: './number-with-units-interaction.component.html',
+  styleUrls: []
+})
+export class InteractiveNumberWithUnitsComponent
+  implements OnInit, OnDestroy {
+  @Input() labelForFocusTarget: string;
+  @Input() savedSolution: InteractionAnswer;
+  componentSubscriptions: Subscription = new Subscription();
+  FORM_ERROR_TYPE: string = 'NUMBER_WITH_UNITS_FORMAT_ERROR';
+  errorMessage: string = '';
+  answer: string = '';
+  isValid: boolean = true;
+  answerChanged: Subject<string> = new Subject<string>();
+  NUMBER_WITH_UNITS_FORM_SCHEMA = {
+    type: 'unicode',
+    ui_config: {}
+  };
+
+  constructor(
+    private currentInteractionService: CurrentInteractionService,
+    private focusManagerService: FocusManagerService,
+    private numberWithUnitsObjectFactory: NumberWithUnitsObjectFactory,
+    private numberWithUnitsRulesService: NumberWithUnitsRulesService,
+    private ngbModal: NgbModal,
+  ) {
+    this.componentSubscriptions.add(this.answerChanged.pipe(
+      // Wait 150ms after the last event before emitting last event.
+      debounceTime(150),
+      // Only emit if value is different from previous value.
+      distinctUntilChanged()
+    ).subscribe(newValue => {
+      try {
+        this.numberWithUnitsObjectFactory.fromRawInputString(newValue);
+        this.errorMessage = '';
+        this.isValid = true;
+      } catch (parsingError) {
+        this.errorMessage = parsingError.message;
+        this.isValid = false;
+      }
+      this.currentInteractionService.updateViewWithNewAnswer();
+    }));
+  }
+
+  ngOnInit(): void {
+    if (this.savedSolution !== undefined) {
+      let savedSolution = this.savedSolution;
+      savedSolution = this.numberWithUnitsObjectFactory.fromDict(
+        savedSolution as NumberWithUnitsAnswer).toString();
+      this.answer = savedSolution;
+    } else {
+      this.answer = '';
     }
-  ]
-});
+
+    try {
+      this.numberWithUnitsObjectFactory.createCurrencyUnits();
+    } catch (parsingError) {}
+
+    const submitAnswerFn = () => this.submitAnswer();
+    const isAnswerValid = () => this.isAnswerValid();
+    this.currentInteractionService.registerCurrentInteraction(
+      submitAnswerFn, isAnswerValid);
+
+    setTimeout(
+      () => {
+        let focusLabel: string = this.labelForFocusTarget;
+        this.focusManagerService.setFocusWithoutScroll(focusLabel);
+      }, 0);
+  }
+
+  submitAnswer(): void {
+    try {
+      var numberWithUnits = (
+        this.numberWithUnitsObjectFactory.fromRawInputString(this.answer));
+      this.currentInteractionService.onSubmit(
+        numberWithUnits as unknown as string,
+        this.numberWithUnitsRulesService as unknown as InteractionRulesService);
```

Comment:
Please try to avoid explicit type conversation.

Reply:
It is necessary, as argument of type `NumberWithUnits` is not assignable to parameter of type 'string'. So first convert it into unknown then using it convert into string.

Reply:
Done